src/components/va-radio/va-radio.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Listen,
+  Prop,
+  h,
+} from '@stencil/core';
+import { getSlottedNodes } from '../../utils/utils';
+
+@Component({
+  tag: 'va-radio',
+  styleUrl: 'va-radio.css',
+  shadow: true,
+})
+export class VaRadio {
+  @Element() el: any;
+
+  @Prop() label: string;
+
+  @Prop() required: boolean = false;
+
+  @Prop() error: string;
+
+  @Prop() enableAnalytics: boolean;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaValueChange: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaKeyDown: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaMouseDown: EventEmitter;
+
+  @Listen('radioOptionSelected')
+  radioOptionSelectedHandler(event: CustomEvent): void {
+    const clickedItem = event.target as HTMLVaRadioOptionElement;
+
+    getSlottedNodes(this.el, 'va-radio-option')
+      .filter(item => item !== clickedItem)
+      .forEach(item => ((item as HTMLVaRadioOptionElement).checked = false));
+
+    clickedItem.checked = true;
```

Comment:
Because each `<input>` is within its own shadow DOM, we lose the benefit of the browser being able to automatically connect them by `name`. This ensure that when one option is checked, the others are unchecked.